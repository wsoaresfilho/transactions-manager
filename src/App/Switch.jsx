import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import { teal, grey } from '@material-ui/core/colors';
import FormGroup from '@material-ui/core/FormGroup';
import Switch from '@material-ui/core/Switch';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';

const PurpleSwitch = withStyles({
    switchBase: {
        color: teal[300],
        '&$track': {
            backgroundColor: teal[100],
        },
        '&$checked': {
            color: teal[800],
        },
        '&$checked + $track': {
            backgroundColor: grey[800],
        },
    },
    checked: {},
    track: {},
})(Switch);

export default function CustomizedSwitches(props) {
    const [customsProps, setCustomState] = React.useState(props);

    React.useEffect(() => {
        setCustomState(props);
    }, [props]);

    const handleChange = () => {
        customsProps.onChange();
    };

    return (
        <FormGroup>
            <Typography component='div'>
                <Grid
                    component='label'
                    container
                    alignItems='center'
                    spacing={1}
                >
                    <Grid item>Light Theme</Grid>
                    <Grid item>
                        <PurpleSwitch
                            checked={customsProps.isChecked}
                            onChange={() => handleChange()}
                        />
                    </Grid>
                    <Grid item>Dark Theme</Grid>
                </Grid>
            </Typography>
        </FormGroup>
    );
}
