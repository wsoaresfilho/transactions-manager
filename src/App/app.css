.app-container {
  max-width: var(--container-max-width);
  margin: auto;
  padding: 0 15px;
}
