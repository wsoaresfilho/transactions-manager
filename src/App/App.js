import React from 'react';
import Header from '../components/Header/Header.jsx';
import RegisterFormContainer from '../components/RegisterTransaction/RegisterTransactionContainer';
import './app.css';

function App() {
    return (
        <div className='app'>
            <Header />
            <div className='app-container'>
                <RegisterFormContainer />
            </div>
        </div>
    );
}

export default App;
