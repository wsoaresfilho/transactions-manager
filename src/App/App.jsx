import React from 'react';
import Header from '../components/Header/Header';
import RegisterTransactionContainer from '../components/RegisterTransaction/RegisterTransactionContainer';
import './app.css';

function App() {
    return (
        <div className='app'>
            <Header />
            <div className='app-container'>
                <RegisterTransactionContainer />
            </div>
        </div>
    );
}

export default App;
