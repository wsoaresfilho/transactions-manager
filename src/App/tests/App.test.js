import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

describe('App component', () => {
    let wrapper;

    beforeAll(() => {
        wrapper = shallow(<App />);
        // console.log(wrapper.debug())
    });

    describe('should render', () => {
        it('app element', () => {
            expect(wrapper.find('.app').length).toBe(1);
        });

        it('Header component', () => {
            expect(wrapper.find('Header').length).toBe(1);
        });

        it('app-container element', () => {
            expect(wrapper.find('.app-container').length).toBe(1);
        });

        it('RegisterTransactionContainer', () => {
            expect(
                wrapper.find('Connect(RegisterTransactionContainer)').exists()
            ).toBeTruthy();
        });

        it('snapshot correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
