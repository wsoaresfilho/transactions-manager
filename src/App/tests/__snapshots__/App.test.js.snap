// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render snapshot correctly 1`] = `ShallowWrapper {}`;
